tmd/Topology/persistent_properties.py

```diff
@@ -0,0 +1,178 @@
+        self._radii = _section_mean_radii(0.5 * tree.d, section_begs, section_ends)
+
+    def get(self, component_start):
+        """
+        Args:
+            component_start (int): The component start
+        Returns:
+            component_angles (list): A list of 1 radius
+        """
+        return [self._radii[component_start]]
+
+    def infinite_component(self, component_start):
+        """Returns mean radius corresponding to inf
+        component
+        """
+        return self.get(component_start)
+
+
+def _phi_theta(u, v):
+    """Computes the angles between vectors u, v
+    in the plane x-y (phi angle) and the plane x-z (theta angle).
+    Returns phi, theta
+
+    Args:
+        u (np.ndarray): (3,) First vector
+        v (np.ndarray): (3,) Second vector
+
+    Returns:
+        delta_phi (float): Difference of phi_angles phi_v - phi_u
+            on the x-y plane
+        delta_theta (float): Difference of theta_angles th_v - th_u
+            on the x-z plane
+    """
+    phi1 = np.arctan2(u[1], u[0])
+    # pylint: disable=assignment-from-no-return
+    theta1 = np.arccos(u[2] / np.linalg.norm(u))
+
+    # pylint: disable=assignment-from-no-return
+    phi2 = np.arctan2(v[1], v[0])
+    theta2 = np.arccos(v[2] / np.linalg.norm(v))
+
+    delta_phi = phi2 - phi1  # np.abs(phi1 - phi2)
+    delta_theta = theta2 - theta1  # np.abs(theta1 - theta2)
+
+    return delta_phi, delta_theta  # dphi, dtheta
+
+
+def _angles_tree(tree, parID, parEND, ch1ID, ch2ID):
+    '''Computes the x-y and x-z angles between parent
+       and children within the given tree.
+
+    Args:
+        tree (Tree): Morphology tree
+        parID (int): Id of parent section
+        parEnd (int): Id ofparent section end
+        ch1ID (int): ID of first child
+        ch2ID (int): ID of section child
+
+    Returns:
+        list:
+            dphi (float): Absolute ifference of phi_angles between parent and first
+
+            dtheta (float)
+            delta_phi (float): Difference of phi_angles phi_v - phi_u
+                on the x-y plane
+            delta_theta (float): Difference of theta_angles th_v - th_u
+                on the x-z plane
+    '''
+
+    parent_direction = tree.get_direction_between(start_id=parID, end_id=parEND)
+    child1_direction = tree.get_direction_between(start_id=parEND, end_id=ch1ID)
+    child2_direction = tree.get_direction_between(start_id=parEND, end_id=ch2ID)
+
+    phi1, theta1 = _phi_theta(parent_direction, child1_direction)
+    phi2, theta2 = _phi_theta(parent_direction, child2_direction)
+
+    if np.abs(phi1) < np.abs(phi2):
+        dphi = phi1
+        dtheta = theta1
+        delta_phi, delta_theta = _phi_theta(child1_direction, child2_direction)
+    else:
+        dphi = phi2
+        dtheta = theta2
+        delta_phi, delta_theta = _phi_theta(child2_direction, child1_direction)
+
+    return [dphi, dtheta, delta_phi, delta_theta]
+
+
+def get_angles(tree, beg, parents, children):
```

Comment:
To be a method, PersistentAngles need to store the tree, which is not the case here. For a function to make sense as a class method, it needs to depend on the internal state of the object. Otherwise, an easy-to-test function can do the job using a clean signature.

Reply:
`@staticmethod` can make it clear that the function is only used in the context of this class and still not depend on any internal state (and it is as easy to test as a plain function). It is just to organize the code with exact same features and signatures.

Reply:
I will change it, but I have to admit don't consider this easier to read or cleaner. `PersistentAngles._phi_theta` will need to be added in angles_tree for example.

Reply:
Indeed, there are some drawbacks. The main point of static methods (in my opinion) is to clarify the code for new comers. When someone reads a new code, static methods immediately tell him that these methods are strongly related to their class but do not depend on the internal state (it can be internal calculations, as in your case, or utilities to format some outputs or stuff like that).
But of course, if you think it is too heavy you can keep it outside the class, that was just a suggestion.

Reply:
I will keep the staticmethods because I agree that it organizes the code a bit better.

Reply:
Ok :)